### PR TITLE
fix: EXPOSED-112 SchemaUtils fails to compare default CURRENT_TIMESTAMP

### PR DESCRIPTION
**Pull request by bog-walk**
If a `timestamp()` or `datetime()` column is created with `CurrentTimestamp()` as the default expression, verifying the table using `addMissingColumnsStatements()` incorrectly generates an alter column statement.

This is because `CurrentTimestamp()` extends `Expression<T>`, so it is [not caught by](https://github.com/JetBrains/Exposed/blob/main/exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/Default.kt#L136) `processForDefaultValue()` and retains parentheses. This results in a comparison mismatch, e.g. in PostgreSQL, between `CURRENT_TIMESTAMP` (from metadata) and `(CURRENT_TIMESTAMP)` (from Exposed).

This issue was partially fixed by [PR #1642 ](https://github.com/JetBrains/Exposed/pull/1642), but `CurrentTimestamp()` was not covered because it does not extend `Function` like the other date/time functions.

**Choosing a columnType:**

The rationale behind why `CurrentTimestamp()` isn't like its counterparts `CurrentDate` and `CurrentDatetime` may be related to the need to provide a specific column type to the `Function` class. Unlike the other 2 functions, `CurrentTimestamp()` can technically be used/compiled with any column, and what's important is that the chosen type can properly hold the result.

All String functions use `TextColumnType` even though they can be used with any `StringColumnType`.
Functions like `Sum`, which have an unknown-sized result, take a `columnType` argument provided by the expression/column on which the extension function is called. 

`CurrentTimestamp()` doesn't have a way to know the type of the column it's being used with, unless the user provides it directly, so the following is an alternative if there are concerns over restricting the function type:
```kt
class CurrentTimestamp<T : Temporal>(
    columnType: IColumnType = JavaInstantColumnType.INSTANCE
) : Function<T>(columnType) { }
```

### CODE REVIEW

File: exposed-java-time/src/main/kotlin/org/jetbrains/exposed/sql/javatime/JavaDateFunctions.kt [hunk 1]
```diff
@@ -49,7 +49,7 @@ object CurrentDate : Function<LocalDate>(JavaLocalDateColumnType.INSTANCE) {
     }
 }
 
-class CurrentTimestamp<T : Temporal> : Expression<T>() {
+class CurrentTimestamp<T : Temporal> : Function<T>(JavaInstantColumnType.INSTANCE) {
```
Comment (bog-walk) [inline exposed-java-time/src/main/kotlin/org/jetbrains/exposed/sql/javatime/JavaDateFunctions.kt:52]: As mentioned in the PR discussion, there may be edge cases not exhausted by the current test suites that arise from restricting the function column type.

`CurrentTimestamp()` doesn't have a way of knowing the type of the column it's being used with, unless the user provides it directly, so the following is an alternative if there are concerns:
```kt
class CurrentTimestamp<T : Temporal>(
    columnType: IColumnType = JavaInstantColumnType.INSTANCE
) : Function<T>(columnType) { }
```

The column type chosen is possibly more a concern for functions that are commonly stand-alone and can be used to create their own column of generated values. If this is being used a default function that returns a timestamp literal, retrieving DB values will be dependent on the type of the column on which `defaultExpression()` is called.

**Review by e5l — APPROVED** [review]

**Review by joc-a — APPROVED** [review]
